tokio/src/sync/mutex.rs

```diff
@@ -340,6 +340,7 @@ impl<T: ?Sized> Mutex<T> {
             let location = std::panic::Location::caller();
 
             tracing::trace_span!(
```

Comment:
This change is necessary, but—IMO—not sufficient. I think we should probably also add a `follows_from` relationship to any resource spans (in contradiction of https://github.com/tokio-rs/tokio/issues/5792) since it doesn't have any impact on memory, leaks or things of that nature. I can't make the following a suggestion because the GitHub only allows me to leave suggestions on _what changed_, but please treat the following as a suggestion:

```rust
            let span = tracing::trace_span!(
                parent: None,
                "runtime.resource",
                concrete_type = "Mutex",
                kind = "Sync",
                loc.file = location.file(),
                loc.line = location.line(),
                loc.col = location.column(),
            );
            span.follows_from(Span::current());
            span
```

(note that [this](https://docs.rs/tracing/latest/tracing/struct.Span.html#method.current) is `Span::current()`. You might need to import it.)

Reply:
That makes sense, right now it would be a no root span in the "void". I'll make that change, thank you.

Reply:
Oh and also, I think it would be more something like this:

```rust
           let current_span = tracing::Span::current();
            let span = tracing::trace_span!(
                parent: None,
                "runtime.resource",
                concrete_type = "Mutex",
                kind = "Sync",
                loc.file = location.file(),
                loc.line = location.line(),
                loc.col = location.column(),
            );
            span.follows_from(current_span);
            span
```

Since `Span::current()` called there will be equal to the trace span we just created.

Reply:
Confirmed through [this test](https://github.com/tokio-rs/tokio/pull/6107/files#diff-47b62efd1013ab2926c772c27e5918a5ce58e8f2e65e0821f2050194467b2a83R193). It does require a fairly large setup, so I'm happy to wait for @hds's`tracing-mock` integration, depending on what you folks would prefer.

Reply:
Ah, I assumed that that the current span wouldn't change until it was entered. whoops.